examples/gno.land/r/gov/dao/v3/impl/prop_requests.gno

```diff
@@ -0,0 +1,216 @@
+package impl
+
+import (
+	"std"
+
+	"gno.land/p/demo/ufmt"
+	"gno.land/r/gov/dao/proxy"
+	"gno.land/r/gov/dao/v3/memberstore"
+)
+
+type ChangeLawMetadata struct {
+	Law *Law
+}
+
+func (a *ChangeLawMetadata) IsMetadata() {}
+func (a *ChangeLawMetadata) String() string {
+	return ufmt.Sprintf("A new Law is proposed:\n %v", a.Law)
+}
+
+func changeLawCallback(m proxy.Metadata) error {
+	amm, ok := m.(*ChangeLawMetadata)
+	if !ok {
+		panic("metadata type for changing the law is wrong")
+	}
+
+	law = amm.Law
+
+	return nil
+}
+
+func NewChangeLawRequest(law *Law) proxy.ProposalRequest {
+	member, _ := memberstore.Get().GetMember(std.OriginCaller())
+	if member == nil {
+		panic("proposer is not a member")
+	}
+
+	return proxy.ProposalRequest{
+		Title:       "Change Law Proposal",
+		Description: "This prpoposal is looking to change the actual govDAO Law",
+		Callback:    changeLawCallback,
+		Metadata: &ChangeLawMetadata{
+			Law: law,
+		},
+	}
+}
+
+type AddMemberMetadata struct {
+	Member    *memberstore.Member
+	Portfolio string
+	Tier      string
+	Addr      std.Address
+	Caller    *memberstore.Member
+}
+
+func (a *AddMemberMetadata) IsMetadata() {}
+func (a *AddMemberMetadata) String() string {
+	return ufmt.Sprintf("A new member with address %v is proposed to be on tier %v. Provided Portfolio information:\n\n%v", a.Addr, a.Tier, a.Portfolio)
+}
+
+func addMemberCallback(m proxy.Metadata) error {
+	amm, ok := m.(*AddMemberMetadata)
+	if !ok {
+		panic("metadata type for adding a member is wrong")
+	}
+
+	amm.Caller.RemoveInvitationPoint()
+
+	return memberstore.Get().SetMember(amm.Tier, amm.Addr, amm.Member)
+}
+
+func NewAddMemberRequest(addr std.Address, tier string, portfolio string) proxy.ProposalRequest {
+	_, ok := memberstore.Tiers.GetTier(tier)
+	if !ok {
+		panic("provided tier does not exists")
+	}
+
+	if tier != memberstore.T1 && tier != memberstore.T2 {
+		panic("Only T1 and T2 members must be added by proposal. To add a T3 member use AddMember function directly.")
+	}
+
+	if portfolio == "" {
+		panic("A portfolio for the proposed member is required")
+	}
+
+	member, _ := memberstore.Get().GetMember(std.OriginCaller())
+	if member == nil {
+		panic("proposer is not a member")
+	}
+
+	if member.InvitationPoints <= 0 {
+		panic("proposer does not have enough invitation points for inviting new people to the board")
+	}
```

Comment:
Should invitation points also be considered when adding a T1 or T2 member?

Reply:
Same question, but I don't get it either when I read the specs directly.

Reply:
I wasn't sure about that neither.